build-system/tasks/compile-jison.js

```diff
@@ -39,23 +39,26 @@ const imports = new Map([
  * Builds parsers for extensions with *.jison files.
  * Uses jison file path to name the parser to export.
  * For example, css-expr-impl.jison creates `cssParser`.
+ *
+ * @param {string} searchDir - directory to compile jison files within.
  * @return {!Promise<void>}
  */
-async function compileJison() {
-  fs.mkdirSync('build/parsers', {recursive: true});
+async function compileJison(searchDir = jisonPath) {
   const startTime = Date.now();
-  const promises = [];
-  jisonPaths.forEach((jisonPath) => {
-    globby.sync(jisonPath).forEach((jisonFile) => {
-      const jsFile = path.basename(jisonFile, '.jison');
-      const extension = jsFile.replace('-expr-impl', '');
-      const parser = extension + 'Parser';
-      const newFilePath = `build/parsers/${jsFile}.js`;
-      promises.push(compileExpr(jisonFile, parser, newFilePath));
-    });
+  fs.mkdirSync('build/parsers', {recursive: true});
+
+  const promises = globby.sync(searchDir).map((jisonFile) => {
+    const jsFile = path.basename(jisonFile, '.jison');
+    const extension = jsFile.replace('-expr-impl', '');
+    const parser = extension + 'Parser';
+    const newFilePath = `build/parsers/${jsFile}.js`;
+    return compileExpr(jisonFile, parser, newFilePath);
   });
   await Promise.all(promises);
-  endBuildStep('Compiled Jison parsers into', 'build/parsers/', startTime);
+
+  if (promises.length > 0) {
+    endBuildStep('Compiled Jison parsers into', 'build/parsers/', startTime);
+  }
```

Comment:
Now that this function is being called once per extension that needs this, I don't htink this message is adding much value. Also, step timing is now rolled into the timing for `buildExtension`. So I think we should delete this block (and `startTime` above). WDYT?

Reply:
It is only printed for extensions that have a parser, but I still agree with you. 
We can keep the utility if we only output the message in case of default arg (build all parsers for whole repo). This only happens for unit, build, and dist.

Reply:
I assume I'm right in saying we no longer need to call this at the start for `build` and `dist` because they call into `buildExtension`, which is now taking care of business. And for `unit`, we don't really care about step timing like we do during compilation (provided things are generally fast).

I'd say skip the logging altogether and revisit later if this ever becomes a long-running thing. WDYT?

Reply:
Done!